bin/ert.in

```diff
@@ -1,73 +1,102 @@
-
-options, args = parser.parse_known_args( args = sys.argv[1:] )
-
-if options.gui:
-    runGui( args )
-elif options.tui:
-    runTui(args)
-elif options.shell:
-    runShell( args )
-elif options.cli:
-    runCli(args)
-else:
-    runDefault(args)
-
+    exec_path = os.path.join(os.path.dirname(__file__), "ert_cli")
+    runExec(exec_path, [args.config, args.algorithm, args.target_case])
+
+
+def ert_parser():
+    parser = ArgumentParser(description="ERT - Ensemble Reservoir Tool")
+
+    subparsers = parser.add_subparsers(
+        title="Available user entries",
+        description='ERT can be accessed through various interfaces. Include '
+                    'one of the following arguments to change between the '
+                    'interfaces. Note that different entry points may require'
+                    'different additional arguments. See the help section for '
+                    'each interface for more details',
+        help="Available entry points")
+
+
+    gui_parser = subparsers.add_parser('gui',
+            help='Graphical User Interface - opens up an independent window for '
+                'the user to interact with ERT.',
+            description='Graphical User Interface')
+    gui_parser.add_argument('config', type=valid_file,
+            help="Ert configuration file")
+    gui_parser.set_defaults(func=runGui)
+
+
+    tui_parser = subparsers.add_parser('text',
+            help='Text user interface - provides a user interface in the terminal.'
+            ' Allows the user to interact within the terminal window ',
+            description='Text user interface')
+    tui_parser.add_argument('config', type=valid_file,
+            help="Ert configuration file")
+    tui_parser.set_defaults(func=runTui)
+
+
+    shell_parser = subparsers.add_parser('shell',
+            help='Shell interface - provides a user interface in the terminal.',
+            description='Shell interface')
+    shell_parser.add_argument('config', type=valid_file,
+            help="Ert configuration file")
+    shell_parser.set_defaults(func=runShell)
+
+
+    cli_parser = subparsers.add_parser('cli',
+            description="Command Line Interface",
+            help="command line help")
+    cli_parser.add_argument('config', type=valid_file,
+            help="Ert configuration file")
+    cli_parser.add_argument("--algorithm", type=str, required=True,
+            choices=["Test-run", "Ensemble Smoother", "Ensemble Experiment"],
+            help="The available algorithms")
+    cli_parser.add_argument("--target-case", type=str, default="default",
+            help="The target filesystem to store results")
+    cli_parser.set_defaults(func=runCli)
+    return parser
+
+
+def main():
+    parser = ert_parser()
+    args = parser.parse_args(sys.argv[1:])
+    args.func(args)
+
+
+if __name__ == '__main__':
+    os.environ["ERT_SHARE_PATH"] = ERT_SHARE_PATH
+    os.environ["ERT_UI_MODE"] = "gui"
+    os.environ["ERT_ROOT"] = ERT_ROOT
```

Comment:
Maybe not necessary anymore?

Reply:
The ERT_ROOT is used internally, so for now we have to keep it